### PR TITLE
451006-MaksimBarouski

### PR DESCRIPTION
**Pull request by maximka777**

### CODE REVIEW

File: ConsoleApp/ConsoleApp/JsonClassInfoReader.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using DTOGeneratorLibrary;
+
+namespace ConsoleApp
+{
+    public class JsonClassInfoReader
+    {
+        public static List<ClassInfo> ReadClassInfoFromFile(string filename)
+        {
+            List<ClassInfo> result;
+            JsonReader reader = new JsonTextReader(new StreamReader(filename));
```
Comment (RamanBut-Husaim) [inline ConsoleApp/ConsoleApp/JsonClassInfoReader.cs]: @maximka777 Disposable objects.

File: ConsoleApp/ConsoleApp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace ConsoleApp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Generating...");
+            try {
+                /*string pathToDir = args[2];
```
Comment (RamanBut-Husaim) [inline ConsoleApp/ConsoleApp/Program.cs]: @maximka777 Remove dead code.

File: ConsoleApp/ConsoleApp/Program.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                string pathToFile = args[0];
+                int maxThreadCount = Int32.Parse(args[1]);*/
+
+                string pathToDir = "generated_classes";
```
Comment (RamanBut-Husaim) [inline ConsoleApp/ConsoleApp/Program.cs]: @maximka777 To command line args.

File: ConsoleApp/DTOGeneratorLibrary/TypeService.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace DTOGeneratorLibrary
+{
+    class TypeService
```
Comment (RamanBut-Husaim) [inline ConsoleApp/DTOGeneratorLibrary/TypeService.cs]: @maximka777 Explicit visibility modifiers.

File: ConsoleApp/DTOGeneratorLibrary/TypeService.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            if(result != null)
+            {
+                return result;
+            }
+            result = TryGetFromForeignTypeConverters(typeInfo);
+            if (result != null)
+            {
+                return result;
+            }
+            return "";
```
Comment (RamanBut-Husaim) [inline ConsoleApp/DTOGeneratorLibrary/TypeService.cs]: @maximka777 `string.Empty`

File: ConsoleApp/MultiThreading/ThreadPool.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MultiThreading
+{
+    public class ThreadPool<I, O>
```
Comment (RamanBut-Husaim) [inline ConsoleApp/MultiThreading/ThreadPool.cs]: @maximka777 As you wrap entity that implements `IDisposable` do not forget to clean up resources. [Take a look](http://www.codeproject.com/Articles/15360/Implementing-IDisposable-and-the-Dispose-Pattern-P)

**Comment by maximka777** [comment]
fixed over all comments

**Pull request merged**